src/plugins/intel_gpu/src/plugin/transformations/swiglu_fusion.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "swiglu_fusion.hpp"
+
+#include "intel_gpu/op/swiglu.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/swish.hpp"
+#include "openvino/op/variadic_split.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+SwiGLUFusion::SwiGLUFusion() {
+    using namespace ov::pass::pattern;
+
+    auto last_dim_static = [](const ov::Output<ov::Node>& output) {
+        auto out_ps = output.get_node()->get_output_partial_shape(0);
+        return out_ps.rank().is_static() && out_ps[out_ps.rank().get_length() - 1].is_static() && out_ps.size() <= 5;
+    };
+
+    // Detect SwiGLU decomposition pattern
+    // SwiGLU(Xw, Xv, beta) = (Xw * (1.0 + exp(-beta * Xw))) * Xv
+    auto data_m = any_input(last_dim_static);
+
+    // VariadicSplit(X, axis, split_lengths) = Xw, Xv
+    auto axis_const_m = wrap_type<ov::op::v0::Constant>();
+    auto split_lengths_const_m = wrap_type<ov::op::v0::Constant>();
+    auto variadic_split_m = wrap_type<ov::op::v1::VariadicSplit>({data_m, axis_const_m, split_lengths_const_m});
+    variadic_split_m->set_output_size(2);
+
+    // Swish(Xw) = Xw * (1.0 + exp(-beta * Xw))
+    auto swish_m = wrap_type<ov::op::v4::Swish>({variadic_split_m->output(0)});
+
+    // Mul(Xw, Xv) = Swish(Xw) * Xv
+    auto mul_m = wrap_type<ov::op::v1::Multiply>({swish_m, variadic_split_m->output(1)});
+
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        OPENVINO_ASSERT(pattern_map.count(mul_m));
+        OPENVINO_ASSERT(pattern_map.count(swish_m));
+        OPENVINO_ASSERT(pattern_map.count(variadic_split_m));
+        OPENVINO_ASSERT(pattern_map.count(split_lengths_const_m));
+        OPENVINO_ASSERT(pattern_map.count(axis_const_m));
+        auto mul = std::dynamic_pointer_cast<ov::op::v1::Multiply>(pattern_map.at(mul_m).get_node_shared_ptr());
+        if (!mul || transformation_callback(mul))
+            return false;
+
+        if (mul->input_value(1).get_index() != 1)
+            return false;
+        auto swish = std::dynamic_pointer_cast<ov::op::v4::Swish>(pattern_map.at(swish_m).get_node_shared_ptr());
+        if (swish->input_value(0).get_index() != 0)
+            return false;
```

Comment:
Are these port id checks really needed? I think pattern matcher should guarantee that.

Reply:
removed